ncdfchecker.py

```diff
@@ -189,16 +193,66 @@ def check_globals(product, constraints, skip=["short_name"], strict=False,
     return errcount, warncount
 
 
-def check_stepsize(data, stepsize):
+def get_month_midpoints(monthslist):
+    """Return a list of dates that correspond to the middle of the month."""
+    def midpoint(start, end): return start + (end - start) / 2
+
+    midmonths = [midpoint(monthslist[i], monthslist[i+1])
+                 for i in range(len(monthslist)-1)]
+
+    return midmonths
+
+
+def get_expected_monthly_stepsize(initdate, runlength):
     """
-    Helper routine to check that the stepsize in some given data is equal to
-    a given value
+    Using the model initialisation date and run length, calculate the
+    expected step size between entries in a monthly file. Use the midpoint
+    in each month to calculate said step size.
+
+    :arg: string initdate : Model initialisation date
+    :arg: int runlength   : Model run length in days
+
+    """
+    SECS_IN_HR = 60 * 60
+
+    initdate = dateutil.parser.parse(initdate)
+    enddate = initdate + datetime.timedelta(days=runlength)
+
+    # Get a list of monthly dates that correspond to whole months that
+    # have been covered by the model run.
+    complete_months = list(
+        dateutil.rrule.rrule(dateutil.rrule.MONTHLY, bymonthday=1,
+                             dtstart=initdate, until=enddate)
+        )
+
+    midmonths = np.array(get_month_midpoints(complete_months))
+
+    expected_stepsize = midmonths[1:len(midmonths)] - \
+        midmonths[0:len(midmonths)-1]
+
+    return [timedelta.total_seconds() / SECS_IN_HR
+            for timedelta in expected_stepsize]
+
+
+def check_stepsize(data, stepsize, monthly=False):
```

Comment:
Don't tie to specific period type - fixed_period="unset" and insert "monthly" to trigger off your specific checks. If block so hangs off "monthly", "unset", and else to not implemented error at minimum here.